packages/core/rfcs/errors-habdling-in-composables.md

```diff
@@ -0,0 +1,63 @@
+# Errors handling in composables
+
+## Motivation
+
+We need to handle errors, that are caused by external api calls provided by factory params.
+Furthermore, it must be something to recognize where that error comes from (which call).
+
+We can solve it by introducing fields inside the errors object and the `AgnosticError`:
+
+Factory
+```js
+const useProductFactory = (factoryParams) => {
+  const useProduct = () => {
+    const products = ref([]);
+    const errors = ref({};)
+
+
+    const search = async (params) => {
+      try {
+        products.value = factoryParams.searchProducts(params);
+      } catch (err) {
+        errors.searchProducts = factoryParams.convertError(err);
```

Comment:
params should also be passed to the converError in order to supply additional information about context of this method used.

If we execute one method multiple times we will lose all errors except the one from last execution